File: android/app/src/main/kotlin/com/example/a/ImageSaver.kt
package com.example.a

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import java.io.File
import java.io.FileOutputStream

object ImageSaver {
    fun saveImage(data: ByteArray, path: String) {
        try {
            println("[DEBUG] Decoding image data")
            val bitmap = BitmapFactory.decodeByteArray(data, 0, data.size)
            if (bitmap == null) {
                println("[ERROR] Failed to decode image data")
                throw Exception("Failed to decode image data")
            }
            
            println("[DEBUG] Creating file: $path")
            val file = File(path)
            
            // Assicurati che la directory esista
            file.parentFile?.also {
                if (!it.exists()) {
                    println("[DEBUG] Creating parent directories")
                    if (!it.mkdirs()) {
                        println("[WARNING] Failed to create parent directories")
                    }
                }
            }
            
            println("[DEBUG] Opening output stream")
            val out = FileOutputStream(file)
            
            println("[DEBUG] Compressing and saving image")
            bitmap.compress(Bitmap.CompressFormat.JPEG, 100, out)
            
            println("[DEBUG] Flushing and closing output stream")
            out.flush()
            out.close()
            
            println("[DEBUG] Image saved successfully")
        } catch (e: Exception) {
            println("[ERROR] Error saving image: ${e.message}")
            e.printStackTrace()
            throw e
        }
    }
} 

File: ios/Runner/ImageSaver.swift
import Foundation
import UIKit

class ImageSaver {
    static func saveImage(data: Data, path: String) throws {
        NSLog("[DEBUG] Decoding image data")
        
        // Verifica che i dati possano essere convertiti in un'immagine
        guard let image = UIImage(data: data) else {
            NSLog("[ERROR] Failed to decode image data")
            throw NSError(domain: "ImageSaver", code: -1, userInfo: [NSLocalizedDescriptionKey: "Failed to decode image data"])
        }
        
        NSLog("[DEBUG] Creating file: \(path)")
        let fileURL = URL(fileURLWithPath: path)
        
        // Assicurati che la directory esista
        let directory = fileURL.deletingLastPathComponent()
        try FileManager.default.createDirectory(at: directory, withIntermediateDirectories: true, attributes: nil)
        
        // Salva l'immagine come JPEG
        NSLog("[DEBUG] Compressing and saving image")
        guard let jpegData = image.jpegData(compressionQuality: 1.0) else {
            NSLog("[ERROR] Failed to create JPEG data from image")
            throw NSError(domain: "ImageSaver", code: -2, userInfo: [NSLocalizedDescriptionKey: "Failed to create JPEG data from image"])
        }
        
        NSLog("[DEBUG] Writing file to disk")
        try jpegData.write(to: fileURL, options: .atomic)
        
        NSLog("[DEBUG] Image saved successfully")
    }
    
    static func saveImageToPhotosAlbum(data: Data, completion: @escaping (Bool, Error?) -> Void) {
        guard let image = UIImage(data: data) else {
            let error = NSError(domain: "ImageSaver", code: -1, userInfo: [NSLocalizedDescriptionKey: "Failed to decode image data"])
            completion(false, error)
            return
        }
        
        UIImageWriteToSavedPhotosAlbum(image, nil, nil, nil)
        completion(true, nil)
    }
} 